### PR TITLE
Add Marshmallon

### PR DESCRIPTION
**Pull request by Spandamn**

### CODE REVIEW

File: data/mods/ssb/abilities.ts [hunk 1]
```diff
@@ -733,6 +733,70 @@ export const Abilities: {[k: string]: ModdedAbilityData} = {
...
+		onStart(pokemon) {
+			pokemon.side.foe.addSideCondition('stubbornness', pokemon);
+		},
+		onDamage(damage, target, source, effect) {
+			if (effect.id === 'recoil') {
+				if (!this.activeMove) throw new Error("Battle.activeMove is null");
+				if (this.activeMove.id !== 'struggle') return null;
+			}
+		},
+		condition: {
```
Comment (ghost) [inline data/mods/ssb/abilities.ts]: This looks too complicated.

File: data/mods/ssb/random-teams.ts [hunk 2]
```diff
@@ -578,6 +584,13 @@ export class RandomStaffBrosTeams extends RandomTeams {
 			// Any set specific tweaks occur here.
 			if (set.name === 'quadrophenic') set.moves[this.random(2) + 1] = 'Conversion';
 
+			if (set.name === 'Marshmallon') {
+				if (set.moves[1] === set.moves[2]) {
+					const moveChoices = ['Flare Blitz', 'Wood Hammer', 'Head Smash'];
+					set.moves[2] = moveChoices.filter((move: string) => move !== set.moves[1])[this.random(2)];
+				}
+			}
```
Comment (ghost) [inline data/mods/ssb/random-teams.ts]: this is not going to fix the problem you are trying to solve. Just randomly generate all 3 moves and replace one with head charge if it isnt there

File: data/mods/ssb/moves.ts [hunk 3]
```diff
@@ -1566,6 +1566,32 @@ export const Moves: {[k: string]: ModdedMoveData} = {
...
+			this.attrLastMove('[still]');
+		},
+		onPrepareHit(target, source) {
+			this.add('-anim', source, 'Slack Off', source);
+			this.add('-anim', source, 'Roar of Time', target);
+			this.add('-anim', source, 'Roar', target);
+		},
+		forceSwitch: true,
+		selfSwitch: true,
+		heal: [1, 2],
```
Comment (ghost) [inline data/mods/ssb/moves.ts]: this is going to heal the opponent

File: data/mods/ssb/abilities.ts (hunk 1 above)
Comment (ghost) [reply data/mods/ssb/abilities.ts]: this isnt on you I just wanted to say that fwiw

File: data/mods/ssb/moves.ts [hunk 4]
```diff
@@ -1566,6 +1566,35 @@ export const Moves: {[k: string]: ModdedMoveData} = {
...
+			this.attrLastMove('[still]');
+		},
+		onPrepareHit(target, source) {
+			this.add('-anim', source, 'Slack Off', source);
+			this.add('-anim', source, 'Roar of Time', target);
+			this.add('-anim', source, 'Roar', target);
+		},
+		onHit(target, source) {
+			this.heal(Math.ceil(source.maxhp * 0.5), source);
+		},
```
Comment (ghost) [inline data/mods/ssb/moves.ts]:
```suggestion
		onAfterMoveSecondarySelf(pokemon, target, move) {
			this.heal(pokemon.maxhp / 2, pokemon, pokemon, move);
		},
```

File: data/mods/ssb/random-teams.ts (hunk 2 above)
Comment (Spandamn) [reply data/mods/ssb/random-teams.ts]: it should replace the third move if it is same as the second

Comment (ghost) [reply data/mods/ssb/random-teams.ts]: true, but if you rng say, Flare Blitz, Head Charge, Flare Blitz, then that filter wont catch it.

Comment (Spandamn) [reply data/mods/ssb/random-teams.ts]: hmmm okay

**Comment by Spandamn** [comment]
Did

File: data/mods/ssb/random-teams.ts [hunk 5]
```diff
@@ -273,6 +273,12 @@ export const ssbSets: SSBSets = {
 		signatureMove: 'Corrosive Cloud',
 		evs: {hp: 252, def: 252, spd: 4}, nature: 'Bold',
 	},
+	Marshmallon: {
+		species: 'Munchlax', ability: 'Stubbornness', item: 'Eviolite', gender: 'M',
+		moves: ['Head Charge', ['Flare Blitz', 'Wood Hammer', 'Head Smash'], ['Flare Blitz', 'Wood Hammer', 'Head Smash']],
```
Comment (ghost) [inline data/mods/ssb/random-teams.ts]:
```suggestion
		moves: ['Head Charge', 'Flare Blitz', 'Wood Hammer', 'Head Smash'],
```

File: data/mods/ssb/random-teams.ts [hunk 6]
```diff
@@ -578,6 +584,14 @@ export class RandomStaffBrosTeams extends RandomTeams {
 			// Any set specific tweaks occur here.
 			if (set.name === 'quadrophenic') set.moves[this.random(2) + 1] = 'Conversion';
 
+			if (set.name === 'Marshmallon') {
+				const moveChoices = ['Head Charge', 'Flare Blitz', 'Wood Hammer', 'Head Smash'];
+				for (let i = 0; i < set.moves.length - 1; i++) {
+					set.moves[i] = this.sampleNoReplace(moveChoices);
+				}
+				if (!set.moves.includes('Head Charge')) set.moves[this.random(3)] = 'Head Charge';
+			}
```
Comment (ghost) [inline data/mods/ssb/random-teams.ts]:
```suggestion
			if (set.name === 'Marshmallon' && !set.moves.includes('Head Charge')) set.moves[this.random(3)] = 'Head Charge';
```

**Pull request merged**